### PR TITLE
enh(api): add groups property to resources details endpoints

### PR DESCRIPTION
**Pull request by adr-mo**
This PR aims to add a new property to the endpoint resource detail.

The groups property relates to
- Hostgroups for the host resources
- Servicegroups for the service resources

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Add a host to one or several hostgroups
- Add a service to one or several servicegroups

- Check that when calling endpoints` /monitoring/resources/hosts/<hostId>` and `/monitoring/resources/hosts/<hostId>/services/<serviceId>` you get the groups property.

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by sc979 — CHANGES_REQUESTED** [review]

File: src/Centreon/Domain/Monitoring/ResourceGroup.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /**
+     * ResourceGroup constructor
+     */
+    public function __construct(int $resourceGroupId, string $resourceGroupName)
+    {
+        $this->id = $resourceGroupId;
+        $this->name = $resourceGroupName;
+    }
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/Monitoring/ResourceGroup.php:64]: Could you switch this method before getId()  ? And add @param doc block.

Comment (adr-mo) [reply src/Centreon/Domain/Monitoring/ResourceGroup.php]: Yep done :+1:

File: src/Centreon/Domain/Monitoring/ResourceGroup.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @param string $resourceGroupName
+     * @return ResourceGroup
+     */
+    public function setName(string $resourceGroupName): self
+    {
+        $this->name = $resourceGroupName;
+
+        return $this;
+    }
```
[the remark this answers is not in the document]
Comment (adr-mo) [reply src/Centreon/Domain/Monitoring/ResourceGroup.php]: I assume that this relates also to @jeremyjaouen comment

**Review by jeremyjaouen — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Centreon/Domain/Monitoring/Resource.php [hunk 3]
```diff
@@ -904,4 +912,28 @@ public function setAcknowledgement(?Acknowledgement $acknowledgement): self
...
+
+    /**
+     * Set groups to which belongs the resource
+     *
+     * @param ResourceGroup[] $groups
+     * @return Resource
+     */
+    public function setGroups(array $groups): self
+    {
+        $this->groups = $groups;
```
[the remark this answers is not in the document]
Comment (adr-mo) [reply src/Centreon/Domain/Monitoring/Resource.php:922]: done :heavy_check_mark:

File: src/Centreon/Domain/Monitoring/ResourceService.php [hunk 4]
```diff
@@ -165,6 +184,23 @@ public function enrichServiceWithDetails(ResourceEntity $resource): void
                 $resource->setAcknowledgement($acknowledgements[0]);
             }
         }
+
+        /**
+         * Get servicegroups to which belongs the actual service resource.
+         */
+        $serviceGroups = $this->monitoringRepository
+            ->findServiceGroupsByHostAndService($resource->getParent()->getId(), $resource->getId());
```
[the remark this answers is not in the document]
Comment (adr-mo) [reply src/Centreon/Domain/Monitoring/ResourceService.php:194]: done :heavy_check_mark:

**Review by callapa — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**